### ISSUE TITLE
New meeting view updates implemented

### ISSUE THREAD

**Issue by ackernaut** [ParabolInc/parabol#4237]
### Background

This is output from #4013 and supports starting v1 Sprint Poker meetings.

My biggest takeaway is we should have an optimistic UX for creating meetings early. While an async-first UX doesn’t prevent folks from completing a meeting in 1 sync session, a sync-first UX can block the async behaviors and optionality. We do this by reducing the friction folks have expressed at starting a meeting. It should feel okay to open a meeting space, that the meeting doesn’t have to take place right away. Kind of like adding an item to Google calendar.

The second takeaway is that some activities are blocked (adding reflections, adding stories) until a meeting exists. Without major changes to how meetings are created and started, we have few options to set folks up to get started early. Part of this is with our primary actions and with any in-app tips we provide.

In the short term, we can experiment with button copy and tips. In the long term, we can create the meetings on the team space where it’s clear that they are open and what the next primary action is.

**Examples**:
- [Mobile-first layout](https://www.figma.com/file/2CdDxgchV8PHXT8YfWfGpw/Meeting-Lobby?node-id=473%3A2645) with a carousel that shows the other types, with a single vertical column and contextual primary button
- [The new meeting view opens over the team view](https://www.figma.com/file/f2224pwzQgNcRmY19LSgr1/New-Team-Activity?node-id=60%3A787). When the new meeting is created it’s added to the team view. The meeting card has a primary action of what to do next.
- [A Loom demo](https://www.loom.com/share/b827116f7d4542128d951ba7327c0f99) walking through some of the ideas
- More ideas and nuances [in the comment below](https://github.com/ParabolInc/parabol/issues/4237#issuecomment-703736391)

### Acceptance criteria
- [ ] The [floating action button on the dashboard](https://www.figma.com/file/f2224pwzQgNcRmY19LSgr1/New-Team-Activity?node-id=60%3A787) says ‘**+ Create Meeting**’ instead of ‘**Start Meeting**’
- [ ] The [new meeting view shows the other meeting types in the carousel](https://www.figma.com/file/2CdDxgchV8PHXT8YfWfGpw/Meeting-Lobby?node-id=473%3A2645), at least at larger breakpoints. Today you can only see one type at a time
- [ ] We change the template editing icon from a dropdown icon to the pencil icon
- [ ] We consider putting a tooltip on these controls (team, template) since there is not an explicit label
- [ ] The ‘**Start Meeting**’ button label becomes contextual:
  - poker = ‘**Add Stories**’
  - retro = ‘**Open for Reflections**’
  - check-in ‘**Begin Check-In**’
- [ ] There is a tip below the primary button for each type:
  - poker = nudge to open early to add stories before the upcoming meeting
  - retro = nudge to open early for reflections over the course of the sprint
  - check-in = tip about adding to the agenda before or during the meeting

### Estimated effort
- 16 hours

**Comment by ackernaut** [comment]
@mattkrick pulling on you for review

**Comment by ackernaut** [comment]
This is a line in the sand, trying to solve around getting meetings started before they are held by a team.

Today I can open a retro board at the top of a sprint, set a time box, and then meet with my team later. I can include the icebreaker and go back to it. On the other hand, I can hold any phase, even discussion, in async.

With a poker board I can do the same. I can open a meeting, add stories, and then go back to an icebreaker when it’s time to meet with my team.

So there’s not a problem with functionality. It’s about helping folks navigate their expected path forward.

If the ‘Start Meeting’ button becomes ‘Add Stories’ or ‘Open for Reflections’ and the first thing they see is the Icebreaker phase, that’s confusing.

More bad ideas:

What if the Icebreaker toggle hides the Icebreaker phase. In the left nav, instead of an Icebreaker nav item, there’s an equally sized control that says ‘+ Add Icebreaker’ which unhides the phase. What if the control in the new meeting view said ‘Hide Icebreaker (you can add it later)’ instead of ‘Include Icebreaker’ assuming async intent. Folks still get the boolean they want either way.

What if meetings had an explicit pre-work phase. If folks select that, we just move the optional icebreaker to after that phase.

What if in the new meeting view, when poker is selected, there’s a control that says ‘Add Stories’ that looks like the controls for selecting the team or template. Tapping this control creates the meeting and moves them to the Scope phase. If they tap this control we can assume async intent and add a ‘Start Estimation’ button that feels like a pseudo start meeting button. They don’t need it but what if they think they need it, kind of like a Select button for #4128.

What if meetings could be created and later started in certain cases. A created meeting is visible to the person who created it. They can take actions on it to prepare, and then start it, opening it to the team.

- **Create meeting**
- Meeting created. **Add stories**
- Stories added. **Start this meeting** with your team

**Comment by ackernaut** [comment]
Related to #4012

**Comment by ackernaut** [comment]
@mattkrick it’d probably be easiest to jump on a video call for 30-45 minutes to kick some ideas around. Mind sending an invite for a preferred time?

**Comment by mattkrick** [comment]
anytime tomorrow works!

**Comment by jordanh** [comment]
@ackernaut – this is a super fun, super difficult place to explore. If my word salad is too hard to parse we could find some time to draw pictures together. Here goes:

### Stuff I believe in

* In the future, the carousel goes away
* You start all activities from the team view—we have some zero-state thing to train you to do this if you are patient zero
* When you start an activity, it shows up as a card on the team view
* This card can be expanded/scrolled to configure that activity examples:
   * Enable/disable/configure an icebreaker
   * Configure/set up scope phase
   * Pick retro template, add Slack integration, schedule date and time of retro, etc.
* These activity cards have permalinks
* These activities can can be expanded into a full-screen meeting experience when it is time to go sync, probably by accessing the permalink

I think this mock you made is closest to this direction:

![Single tasks kanban view](https://user-images.githubusercontent.com/103000/95272353-f6dc9d00-07f4-11eb-854c-c7d8584b1158.png)

Also:
* A future stand-up/demo thingy is just another activity
* A conversation or decision is also just another activity
* Eventually these things should have id numbers and should be able to relate to each other


# Stuff I am unsure about

* How do we handle the following JTBD: as a scrum master, when it is time to hold our team's retro, I want a link my team can access from a recurring calendar appointment, so I don't have to worry about how to get folks to the right place

* When you expand a card, does it make the card expand or does it just appear in the right nav using an inspector pattern?
